plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/vendored/ParquetConversions.java

```diff
@@ -0,0 +1,112 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.iceberg.vendored;
+
+import org.apache.iceberg.expressions.Literal;
+import org.apache.iceberg.parquet.ParquetUtil;
+import org.apache.iceberg.types.Type;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.LogicalTypeAnnotation.DecimalLogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import java.util.function.Function;
+
+class ParquetConversions
+{
+    private ParquetConversions()
+    {
+    }
+
+    @SuppressWarnings("unchecked")
+    static <T> Literal<T> fromParquetPrimitive(Type type, PrimitiveType parquetType, Object value)
+    {
+        switch (type.typeId()) {
+            case BOOLEAN:
+                return (Literal<T>) Literal.of((Boolean) value);
+            case INTEGER:
+            case DATE:
+                return (Literal<T>) Literal.of((Integer) value);
+            case LONG:
+            case TIME:
+            case TIMESTAMP:
+                return (Literal<T>) Literal.of((Long) value);
+            case FLOAT:
+                return (Literal<T>) Literal.of((Float) value);
+            case DOUBLE:
+                return (Literal<T>) Literal.of((Double) value);
+            case STRING:
+                Function<Object, Object> stringConversion = converterFromParquet(parquetType);
+                return (Literal<T>) Literal.of((CharSequence) stringConversion.apply(value));
+            case UUID:
+                Function<Object, Object> uuidConversion = converterFromParquet(parquetType);
+                return (Literal<T>) Literal.of((UUID) uuidConversion.apply(value));
+            case FIXED:
+            case BINARY:
+                Function<Object, Object> binaryConversion = converterFromParquet(parquetType);
+                return (Literal<T>) Literal.of((ByteBuffer) binaryConversion.apply(value));
+            case DECIMAL:
+                Function<Object, Object> decimalConversion = converterFromParquet(parquetType);
+                return (Literal<T>) Literal.of((BigDecimal) decimalConversion.apply(value));
+            default:
+                throw new IllegalArgumentException("Unsupported primitive type: " + type);
+        }
+    }
+
+    static Function<Object, Object> converterFromParquet(PrimitiveType type)
+    {
+        if (type.getOriginalType() != null) {
+            switch (type.getOriginalType()) {
```

Comment:
Make all of these switch expressions

Reply:
Not possible due to the fallthrough on line 95. Switched other switches to switch expressions.